app/src/main/java/com/vitorpamplona/amethyst/ui/screen/ChatroomListFeedView.kt

```diff
@@ -77,19 +87,70 @@ private fun FeedLoaded(
 ) {
     val listState = rememberLazyListState()
 
-    LazyColumn(
-        contentPadding = PaddingValues(
-            top = 10.dp,
-            bottom = 10.dp
-        ),
-        state = listState
-    ) {
-        itemsIndexed(state.feed.value, key = { index, item -> if (index == 0) index else item.idHex }) { index, item ->
-            ChatroomCompose(
-                item,
-                accountViewModel = accountViewModel,
-                navController = navController
-            )
+    val accountState by accountViewModel.accountLiveData.observeAsState()
+    val account = accountState?.account
+
+    val context = LocalContext.current.applicationContext
+
+    fun markAllAsRead() {
+        state.feed.value.forEach { note ->
+            val routeForLastRead = if (note.event == null) {
+                return@forEach
+            } else if (note.channel != null) {
+                note.channel?.let {
+                    "Channel/${it.idHex}"
+                }
+
+            } else {
+                account?.let {
+                    val replyAuthorBase = note.mentions?.first()
+
+                    var userToComposeOn = note.author!!
+
+                    if (replyAuthorBase != null) {
+                        if (note.author == account.userProfile()) {
+                            userToComposeOn = replyAuthorBase
+                        }
+                    }
+
+                    "Room/${userToComposeOn.pubkeyHex}"
+                }
+            }
+
+            routeForLastRead?.let {
+                val createdAt = note.event?.createdAt
+                if (createdAt != null) {
+                    NotificationCache.markAsRead(it, createdAt, context)
```

Comment:
Should we consider marking as read with the current time instead of created at?

Reply:
Especially if there's a risk of the the note, and its created at, being outdated.